src/libraries/Native/Unix/System.Globalization.Native/pal_casing.c

```diff
@@ -150,6 +150,24 @@ void GlobalizationNative_ChangeCaseTurkish(
     }
 }
 
+void GlobalizationNative_InitOrdinalCasingPage(int32_t pageNumber, UChar* pTarget)
+{
+    pageNumber <<= 8;
+    for (int i = 0; i < 256; i++)
+    {
+        // Unfortunately, to ensure one-to-one simple mapping we have to call u_toupper on every character.
+        // Using string casing ICU APIs cannot give such results even when using NULL locale to force root behavior.
+        pTarget[i] = (UChar) u_toupper((UChar32)(pageNumber + i));
+    }
+
+    if (pageNumber == 0x0100)
+    {
+        // Disable Turkish I behavior on Ordinal operations
```

Comment:
I know this is just a clean port, but special cases like this that aren't driven by culture data make me nervous.  How do we know this is the only special case and there won't be more in the future?

Reply:
I added these for compatibility reason not for any other reason. I am not really worried if we get any cases in the future as these will be a new introduced character that never existed before and it is ok for us to decide to handle it either way.

Reply:
I forgot to mention, disabling these characters I think not related to the culture that these characters belongs to but related to mapping the character to the ASCII range. so it is not related to the cultures.

Reply:
I see, the comment could use some clarification.  My understanding is that this is treating upper/lowercase the same for both the `Turkish lowercase i` and the `Latin small letter long s`.  In other words: these characters should have no upper-case representation, even if `u_toupper` returns an uppercase value for them.  Is that correct?

Reply:
yes that is correct.